include/rdma/rdma_cma.h

```diff
@@ -538,6 +561,30 @@ int rdma_join_multicast(struct rdma_cm_id *id, struct sockaddr *addr,
 			void *context);
 
 /**
+ * rdma_join_multicast_ex - Joins a multicast group with options.
+ * @id: Communication identifier associated with the request.
+ * @mc_join_attr: Extensive struct containing multicast join parameters.
+ * @context: User-defined context associated with the join request.
+ * Description:
+ *   Joins a multicast group with options. Currently supporting MC join flags.
+ *   The QP will be attached based on the given join flag.
+ *   Join message will be sent according to the join flag.
+ * Notes:
+ *   Before joining a multicast group, the rdma_cm_id must be bound to
+ *   an RDMA device by calling rdma_bind_addr or rdma_resolve_addr.  Use of
+ *   rdma_resolve_addr requires the local routing tables to resolve the
+ *   multicast address to an RDMA device.  The user must call
+ *   rdma_leave_multicast to leave the multicast group and release any
+ *   multicast resources.  The context is returned to the user through
+ *   the private_data field in the rdma_cm_event.
+ * See also:
+ *   rdma_leave_multicast, rdma_bind_addr, rdma_resolve_addr, rdma_create_qp
+ */
+int rdma_join_multicast_ex(struct rdma_cm_id *id,
+			   struct rdma_cm_join_mc_attr_ex *mc_join_attr,
+			   void *context);
```

Comment:
Ok I understand your approach, I agree it can make things easier for the users. 
Regarding your example, using the same function name and adding a variable can break current API so I will have to add a new function, something like rdma_join_multicast_flags(...).